### ISSUE TITLE
Wrapping all functions that return promise to support Meteor

### ISSUE THREAD

**Issue by vlasky** [sidorares/node-mysql2#912]
Hi, I'm the defacto maintainer of the Meteor MySQL packages https://github.com/vlasky/meteor-mysql and https://github.com/vlasky/mysql-live-select.

I have made a number of improvements to the package in my own private fork, but I want to do some refactoring before releasing them publicly.

In short, I am trying to come up with efficient code that automatically adds Meteor's Promise.await() wrappers around every method in node-mysql2 that returns a promise.

The Meteor framework relies on Fibers/Futures and Promise.await() is like an adapter.

Here's an example of how  things need to be done now when using node-mysql2 in Meteor

```
import mysql from 'mysql2';

const pool = mysql.createPool({
  host: 'localhost',
  user: 'root',
  database: 'test',
  waitForConnections: true,
  connectionLimit: 10,
  queueLimit: 0
});

myConnection = Promise.await(mysql.pool.promise().getConnection());
...

myResult = Promise.await(myConnection.execute(query, args));
```

To achieve this goal, I expect it to require a snippet of code that iterates over node-mysql2's prototypes and create a new object with the Promise.await()-wrapped methods, or even add a new method to mysql.pool like mysql.pool.fibers().

Any guidance would be appreciated.

**Comment by sidorares** [comment]
I'd rather do that manually in a separate wrapper package. There are maybe a dozen of top level api methods that need to be wrapped, maybe no need for sophisticated automation

**Comment by vlasky** [comment]
Yes @sidorares, I am just after the required bit of code - I am not looking for it to be incorporated into node-mysql2.

**Comment by sidorares** [comment]
So you'll have something like this:

```js
import mysql from 'mysql2';
import mysqlFutureAdapter from 'mysql2-future'; // example name, might be taken

const pool = mysqlFutureAdapter (mysql2).createPool();
const myConnection  = pool.getConnection();
const result = myConnection.execute(query, args);
```

**Comment by sidorares** [comment]
slightly offtopic, but I'm keen to integrate better live select functionality. There is already binlog support in mysql2 so should be not that difficult

**Comment by sidorares** [comment]
I'd expect your adapter would look similar to promise wrapper code - https://github.com/sidorares/node-mysql2/blob/master/promise.js
